atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -243,26 +246,50 @@ public void compactInternally() {
                 + " Since this can't be automated in your configuration,"
                 + " good practice would be do to occasional offline manual maintenance of rebuilding"
                 + " IOT tables to compensate for bloat. You can contact Palantir Support if you'd"
-                + " like more information.";
+                + " like more information. Underlying error was: {}";
 
         if (config.enableOracleEnterpriseFeatures()) {
             try {
                 conns.get().executeUnregisteredQuery(
                         "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
                                 + " MOVE ONLINE");
             } catch (PalantirSqlException e) {
-                log.error(compactionFailureTemplate + " Underlying error was: {}",
+                log.error(compactionFailureTemplate,
                         tableRef,
                         "(Enterprise Edition that requires this user to be able to perform DDL operations)",
                         e.getMessage());
             } catch (TableMappingNotFoundException e) {
                 throw new RuntimeException(e);
             }
-        } else {
-            log.warn(compactionFailureTemplate,
-                    tableRef,
-                    "(If you are running against Enterprise Edition,"
-                    + " you can set enableOracleEnterpriseFeatures to true in the configuration.)");
+        } else if (config.enableShrinkOnOracleStandardEdition()) {
+            Stopwatch timer = Stopwatch.createStarted();
+            try {
+                Stopwatch shrinkAndCompactTimer = Stopwatch.createStarted();
+                conns.get().executeUnregisteredQuery(
+                        "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
+                                + " SHRINK SPACE COMPACT");
+                log.info("Call to SHRINK SPACE COMPACT on table {} took {} ms.",
+                        tableRef, shrinkAndCompactTimer.elapsed(TimeUnit.MILLISECONDS));
+
+                Stopwatch shrinkTimer = Stopwatch.createStarted();
+                conns.get().executeUnregisteredQuery(
```

Comment:
We should catch issues with `SHRINK SPACE` and `SHRINK SPACE COMPACT` separately. If `COMPACT` completes but the full shrink fails, then the impl could try to complete the shrink during off-peak hours ([link](https://docs.oracle.com/cd/B28359_01/server.111/b28310/schema003.htm#CBBBIADA)).

Otherwise, what benefit do we get by using `COMPACT` and then immediately trying to finish the shrink?

Reply:
How will the IMPL know which tables to shrink? The log lines are sufficient to tell which operation failed. 

The benefit is that locks on the entire table are held for a much shorter duration.

Reply:
If the `PalantirSqlException` contain the command that failed, then that's OK.
Can we assume that deployments will have `info` logging enabled?

Reply:
yes, `PalantirSqlException` will contain the command that failed.